src/InteractiveWindow/Editor/InteractiveWindow.cs

```diff
@@ -1288,36 +1289,13 @@ private void SetActiveCodeToHistory(History.Entry entry)
 
         private void ClearInput()
         {
-            var sourceSpans = _projectionBuffer.CurrentSnapshot.GetSourceSpans();
-            Debug.Assert(sourceSpans.Count > 0);
-
-            // Finds the last primary prompt (standard input or code input).
-            // Removes all spans following the primary prompt from the projection buffer.
-            int i = sourceSpans.Count - 1;
-            while (i >= 0)
+            if (_stdInputStart != null)
             {
-                var sourceSnapshot = sourceSpans[i].Snapshot;
-                if (GetSpanKind(sourceSnapshot) == ReplSpanKind.Prompt || GetSpanKind(sourceSnapshot) == ReplSpanKind.StandardInputPrompt)
-                {
-                    Debug.Assert(i != sourceSpans.Count - 1);
-                    break;
-                }
-
-                i--;
+                _standardInputBuffer.Delete(Span.FromBounds(_stdInputStart.Value, _standardInputBuffer.CurrentSnapshot.Length));
             }
-
-            if (i >= 0)
+            else
             {
-                var sourceSnapshot = sourceSpans[i].Snapshot;
-                if (GetSpanKind(sourceSnapshot) != ReplSpanKind.StandardInputPrompt)
-                {
-                    _currentLanguageBuffer.Delete(new Span(0, _currentLanguageBuffer.CurrentSnapshot.Length));
-                }
-                else
-                {
-                    Debug.Assert(_stdInputStart != null);
-                    _standardInputBuffer.Delete(Span.FromBounds(_stdInputStart.Value, _standardInputBuffer.CurrentSnapshot.Length));
-                }
+                _currentLanguageBuffer.Delete(new Span(0, _currentLanguageBuffer.CurrentSnapshot.Length));
```

Comment:
The secondary prompts are removed in the ProjectionBufferChanged event handler. Added a test to verify.